tests/base.py

```diff
@@ -209,6 +210,18 @@ def ensure_available_balance(self):
                 )
                 break
 
+    @classmethod
+    def setUpClass(cls):
+        """Run the class-level balance ensure (so setUpClass-level creates are safe)."""
+        # run the existing ensure_available_balance logic early
+        cls.ensure_available_balance()
```

Comment:
Would like to add logging to the ensure balance method so we can monitor this over time.  Also consider bumping the check from 1,000 to 10,000 for both comparison and top-up amount.
```suggestion
        if balance.get('currency') == 'usd':
            balance_usd = balance.get('currency', 0)
            LOGGER.info("balance_usd: %s, pending_usd: %s", balance_usd, pending_amount_usd)
            if balance_usd + pending_amount_usd <= 100000:
                stripe_client.PaymentIntent.create(
```